cmd/prom2json/main.go

```diff
@@ -26,27 +28,86 @@ import (
 	"github.com/prometheus/prom2json"
 )
 
+var USAGE = fmt.Sprintf("Usage: %s [[METRICS_URL | METRICS_PATH] | --url METRICS_URL [--cert CERT_PATH --key KEY_PATH | --accept-invalid-cert] | --file METRICS_PATH | --stdin]", os.Args[0])
+
 func main() {
 	cert := flag.String("cert", "", "certificate file")
 	key := flag.String("key", "", "key file")
 	skipServerCertCheck := flag.Bool("accept-invalid-cert", false, "Accept any certificate during TLS handshake. Insecure, use only for testing.")
+	flag.String("url", "", "Read data from an URL (exclusive with positional argument, --stdin and --file)")
+	flag.String("file", "", "Read data from a file (exclusive with positional argument, --url and --stdin)")
+	flag.Bool("stdin", false, "Read data from stdin (exclusive with positional argument, --file and --url)")
```

Comment:
I find these three mutually exclusive flags quite complicated. Do you think they are really required? The only use case I see is if a filename starts with `http://...` etc. But those users could then use stdin redirection.

In lack of new arguments, I would prefer not not have these flags at all.

Reply:
it is more common in unix systems to use somehting like `--file -` to read from stdin. Just saying.

Reply:
But no flag at all is most common. :o)